samza-core/src/test/java/org/apache/samza/serializers/model/serializers/TestIntermediateMessageSerde.java

```diff
@@ -126,4 +132,25 @@ public void testEndOfStreamMessageSerde() {
     assertEquals(de.getTaskName(), taskName);
     assertEquals(de.getVersion(), 1);
   }
+
+  @Test(expected = SamzaException.class)
+  public void testUserMessageSerdeException() {
+    Serde<?> mockUserMessageSerde = mock(Serde.class);
+    when(mockUserMessageSerde.fromBytes(anyObject())).then(new Answer<Object>() {
+      @Override
+      public Object answer(InvocationOnMock invocation) throws Throwable {
+        byte[] bytes = invocation.getArgumentAt(0, byte[].class);
+        if (Arrays.equals(bytes, new byte[]{1, 2})) {
+          throw new SamzaException("User message serde failed to deserialize this message.");
```

Comment:
Since you have control over it, consider using something other than `SamzaException` for this mock and validation, since `SamzaException` is something that gets thrown a lot by core Samza classes.
It's just a nice small thing you can do to help make tests a little more robust, in case someone introduces a bug in the future.

Reply:
Good suggestion. Updated to use IllegalArgumentException.